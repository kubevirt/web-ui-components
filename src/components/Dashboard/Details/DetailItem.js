import React from 'react';
import PropTypes from 'prop-types';
import { OverlayTrigger, Tooltip } from 'patternfly-react';

import { DASHES } from '../../../constants';

export const DetailItem = ({ title, value, isLoading, LoadingComponent }) => {
  const description = value ? (
    <OverlayTrigger
      overlay={<Tooltip id={`tooltip-for-${title}`}>{value}</Tooltip>}
      placement="top"
      trigger={['hover', 'focus']}
      rootClose={false}
    >
      <span>{value}</span>
    </OverlayTrigger>
  ) : (
    DASHES
  );
  return (
    <React.Fragment>
      <dt className="kubevirt-detail__item-title">{title}</dt>
      <dd className="kubevirt-detail__item-value">{isLoading ? <LoadingComponent /> : description}</dd>
    </React.Fragment>
  );
};

DetailItem.defaultProps = {
  value: null,
};

DetailItem.propTypes = {
  title: PropTypes.string.isRequired,
  value: PropTypes.string,
  isLoading: PropTypes.bool.isRequired,
  LoadingComponent: PropTypes.oneOfType([PropTypes.node, PropTypes.func]).isRequired,
};
