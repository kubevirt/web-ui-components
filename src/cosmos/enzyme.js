import createContext from 'react-cosmos-test/enzyme';

import paths from '../../config/paths';

export default function createTestContext(args) {
  args.proxies = args.proxies || []; // workaround of react-cosmos-test bug
  return createContext({
    ...args,
    cosmosConfigPath: `${paths.config}/cosmos.config.js`,
  });
}
