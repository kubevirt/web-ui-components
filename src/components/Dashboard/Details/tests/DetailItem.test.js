import React from 'react';
import { render } from 'enzyme';

import { DetailItem } from '../DetailItem';
import { default as DetailItemFixtures } from '../fixtures/DetailItem.fixture';

const testDetailItem = () => <DetailItem {...DetailItemFixtures[0].props} />;

describe('<DetailItem />', () => {
  it('renders correctly', () => {
    const component = render(testDetailItem());
    expect(component).toMatchSnapshot();
  });
});
