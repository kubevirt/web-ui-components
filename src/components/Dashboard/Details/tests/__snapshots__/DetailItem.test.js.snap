// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetailItem /> renders correctly 1`] = `
Array [
  <dt
    class="kubevirt-detail__item-title"
  >
    title1
  </dt>,
  <dd
    class="kubevirt-detail__item-value"
  >
    <span>
      value1
    </span>
  </dd>,
]
`;
